src/test/java/seedu/address/logic/commands/FindTaskCommandTest.java

```diff
@@ -0,0 +1,90 @@
+package seedu.address.logic.commands;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.logic.Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX;
+import static seedu.address.logic.Messages.MESSAGE_TASKS_LISTED_OVERVIEW;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandFailure;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static seedu.address.testutil.TypicalIndexes.INDEX_FIRST_PERSON;
+import static seedu.address.testutil.TypicalIndexes.INDEX_SECOND_PERSON;
+import static seedu.address.testutil.TypicalPersons.ALICE;
+import static seedu.address.testutil.TypicalPersons.BOB;
+import static seedu.address.testutil.TypicalTasks.getTypicalAddressBookWithTasks;
+
+import java.util.Collections;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+
+/**
+ * Contains integration tests (interaction with the Model) and unit tests for {@code FindTaskCommand}.
+ */
+public class FindTaskCommandTest {
+
+    private Model model = new ModelManager(getTypicalAddressBookWithTasks(), new UserPrefs());
+    private Model expectedModel = new ModelManager(getTypicalAddressBookWithTasks(), new UserPrefs());
+
+    @Test
+    public void equals() {
+        FindTaskCommand findFirstCommand = new FindTaskCommand(INDEX_FIRST_PERSON);
+        FindTaskCommand findSecondCommand = new FindTaskCommand(INDEX_SECOND_PERSON);
+
+        // same object -> returns true
+        assertTrue(findFirstCommand.equals(findFirstCommand));
+
+        // same values -> returns true
+        FindTaskCommand findFirstCommandCopy = new FindTaskCommand(INDEX_FIRST_PERSON);
+        assertTrue(findFirstCommand.equals(findFirstCommandCopy));
+
+        // different types -> returns false
+        assertFalse(findFirstCommand.equals(1));
+
+        // null -> returns false
+        assertFalse(findFirstCommand.equals(null));
+
+        // different person -> returns false
+        assertFalse(findFirstCommand.equals(findSecondCommand));
+    }
+
+    @Test
+    public void execute_invalidPersonIndex_throwsCommandException() {
+        Index outOfBoundsIndex = Index.fromOneBased(model.getFilteredPersonList().size() + 1);
+        FindTaskCommand findTaskCommand = new FindTaskCommand(outOfBoundsIndex);
+
```

Comment:
I like how you implemented boundary testing by checking the behavior when an index is out of bounds. It ensures that the application robustly handles user errors, preventing runtime exceptions.